### ISSUE TITLE
Questions about weight initialization of deformable convolution when self.groups is not equal to 1

### ISSUE THREAD

**Issue by Fu0511** [open-mmlab/mmdetection#5622]
In the code of deformable convolution,  the weight is defined as `torch.Tensor(out_channels, in_channels // self.groups, *self.kernel_size))` : it is a tensor of `out_channels*(in_channels // self.groups)*kernel_size[0]*kernel_size[1]`. Then, in the initialization of the weight, the stdv of Kaiming initialization  is defined as `1/math.sqrt((in_channels*kernel_size[0]*kernel_size[1]))`. 

However, I wonder that **if self.groups is not equal to 1**, the stdv should be `1/math.sqrt((in_channels // self.groups)*kernel_size[0]*kernel_size[1])`. Right?

`class DeformConv(nn.Module):

    def __init__(self,
                 in_channels,
                 out_channels,
                 kernel_size,
                 stride=1,
                 padding=0,
                 dilation=1,
                 groups=1,
                 deformable_groups=1,
                 bias=False):
        super(DeformConv, self).__init__()

        assert not bias
        assert in_channels % groups == 0, \
            'in_channels {} cannot be divisible by groups {}'.format(
                in_channels, groups)
        assert out_channels % groups == 0, \
            'out_channels {} cannot be divisible by groups {}'.format(
                out_channels, groups)

        self.in_channels = in_channels
        self.out_channels = out_channels
        self.kernel_size = _pair(kernel_size)
        self.stride = _pair(stride)
        self.padding = _pair(padding)
        self.dilation = _pair(dilation)
        self.groups = groups
        self.deformable_groups = deformable_groups

        self.weight = nn.Parameter(
            torch.Tensor(out_channels, in_channels // self.groups,
                         *self.kernel_size))

        self.reset_parameters()

    def reset_parameters(self):
        n = self.in_channels
        for k in self.kernel_size:
            n *= k
        stdv = 1. / math.sqrt(n)
        self.weight.data.uniform_(-stdv, stdv)

    def forward(self, x, offset):
        return deform_conv(x, offset, self.weight, self.stride, self.padding,
                           self.dilation, self.groups, self.deformable_groups)
`

>

**Comment by jshilong** [comment]
@ZwwWayne @hhaAndroid 
Is there any suggestion? I am not familiar with this topic

**Comment by ZwwWayne** [comment]
Hi @Fu0511 ,
It seems to be a legacy issue where we rarely use the group version of DCN. But your suggestion is reasonable. Would you like to create a PR in MMCV to fix that? @jshilong  will benchmark the performance before and after the modification.
